addons/product/models/product_uom.py

```diff
@@ -0,0 +1,31 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductUom(models.Model):
+    _description = 'Link between the products and the UoM'
+
+    uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True, ondelete='cascade')
+    product_id = fields.Many2one('product.product', 'Product', required=True, ondelete='cascade')
+    name = fields.Char('Barcode', index='btree_not_null', required=True)
+    company_id = fields.Many2one('res.company', 'Company', required=True, default=lambda self: self.env.company)
```

Comment:
Is a company needed? It seems like this should have the same behavior as the barcodes for products: is unique if a company is set, but is cross company if no company_id set. Also, currently product_packagings don't require a company_id